src/Application.php

```diff
@@ -290,4 +277,61 @@ private function checkAndCompletionInputMapping(array $transformationConfig): ar
         }
         return $transformationConfig;
     }
+
+    private function sortTransformationRows(array $transformationConfig): array
```

Comment:
Nemohla by to byt prosim nejaka trieda, co ja viem `RowsSorter`:
- funkcia `writeOutputRow` by mohla byt potom metoda
- interny stav by sa dal pekne ulozit do nejakych `property`
- a nakoniec by sa vytiahli zoradene `rows`.

Tomu kodu uplne nechapem, ... a aj ked ho teraz nejak pochopim, tak to nabuduce budem musiet studovat znova.
Teda navrhujem spravit z toho vlastnu triedu, viac to podelit do metod a pridat komentare co sa tam deje ... aby aj dalsie generacie vedeli nad tym pracovat :)

Reply:
Inak, zoradzovanie objektov podla `dependencies` ([topological sorting](https://en.wikipedia.org/wiki/Topological_sorting)) sa riesi bezne, tak myslim, ze budu existovat nejake kniznice.

Nasiel som napriklad:
https://github.com/marcj/topsort.php

Hned prvy priklad je IMHO co potrebujeme:
![image](https://user-images.githubusercontent.com/19371734/127046434-f9f04524-39e2-41e6-8b7b-fc1c457ea501.png)


Funguje to na `strings`, takze to v pohode pojde s IDs.

Tj. mohlo by sa iterovat nad vsetkymi `rows` vo faze a pridavat ID + zavislosti do `sorter` a nakoniec iba vytiahnut zoradene IDs.

Je to pravdepodobne to iste co robis aj ty, ale takto by to bolo prehladnejsie a ta kniznica je dobre otestovana.
+ menej prace.

Reply:
@webrouse jj máš pravdu... použil jsem knihovnu kterou odkazuješ a testy nechal... oba kódy dělají to samé, tak nechám použitý takto

Reply:
btw další generace už snad bude mít transformace zmigrované :)

Reply:
jj mas pravdu, ... len fakt som to mal problem dostat do hlavy, ... vdaka za vymenu

Reply:
přitom to byl sofistikovaný a promyšlený kus kódu :-D